frontend/js/src/explore/lb-radio/Prompt.tsx

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+
+import * as React from "react";
+import { useState } from "react";
+
+type PromptProps = {
+  onGenerate: (prompt: string, mode: string) => void;
+  errorMessage: string;
+  initMode: string;
+  initPrompt: string;
+};
+
+function Prompt(props: PromptProps) {
+  const { onGenerate, errorMessage, initMode, initPrompt } = props;
+  const [prompt, setPrompt] = useState<string>(initPrompt);
+  const [mode, setMode] = useState<string>(initMode);
+  const [hideExamples, setHideExamples] = React.useState(false);
+
+  const generateCallbackFunction = React.useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      setHideExamples(true);
+      onGenerate(prompt, mode);
+    },
+    [prompt, onGenerate, mode]
+  );
+
+  const onInputChangeCallback = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const text = event.target.value;
+      setPrompt(text);
+    },
+    []
+  );
+
+  const onModeSelectChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLSelectElement>) => {
+      const text = event.target.value;
+      setMode(text);
+    },
+    []
+  );
+
+  React.useEffect(() => {
+    if (prompt.length > 0) {
+      setHideExamples(true);
+      onGenerate(initPrompt, initMode);
+    }
+  }, [initMode, initPrompt, onGenerate, prompt.length]);
```

Comment:
Since this is only used once when we load the component, we want to use `initPrompt` and not `prompt`:
```suggestion
  React.useEffect(() => {
    if (initPrompt.length > 0) {
      setHideExamples(true);
      onGenerate(initPrompt, initMode);
    }
  }, [initMode, initPrompt, onGenerate]);
```